Development/nmos/streamcompatibility_behaviour.cpp

```diff
@@ -0,0 +1,322 @@
+                                    connection_resource.data[nmos::fields::endpoint_staged] = merged;
+                                });
+
+                                nmos::details::handle_immediate_activation_pending(model, lock, sender_id_type, merged[nmos::fields::activation], gate);
+                            }
+                        }
+                    }
+                    catch (const std::exception& e)
+                    {
+                        slog::log<slog::severities::warning>(gate, SLOG_FLF) << "An exception appeared while updating the state of Sender " << sender_id << ": " << e.what();
+                        continue;
+                    }
+                }
+
+                // find IS-11 Receivers with recently updated "caps" to check whether the active transport file still satisfies them
+                for (const nmos::id& receiver_id : streamcompatibility_receivers_ids)
+                {
+                    try
+                    {
+                        bool updated = false;
+
+                        const std::pair<nmos::id, nmos::type> receiver_id_type{ receiver_id, nmos::types::receiver };
+                        auto receiver = find_resource(node_resources, receiver_id_type);
+                        if (node_resources.end() == receiver) throw std::logic_error("Matching IS-04 receiver not found");
+
+                        updated = most_recent_update < nmos::fields::version(nmos::fields::caps(receiver->data));
+
+                        if (updated)
+                        {
+                            slog::log<slog::severities::info>(gate, SLOG_FLF) << "Receiver " << receiver_id << " has been updated recently and Receiver State is being updated as well";
+
+                            const std::pair<nmos::id, nmos::type> streamcompatibility_receiver_id_type{ receiver_id, nmos::types::receiver };
+                            auto streamcompatibility_receiver = find_resource(streamcompatibility_resources, streamcompatibility_receiver_id_type);
+                            if (streamcompatibility_resources.end() == streamcompatibility_receiver) throw std::logic_error("Matching IS-11 receiver not found");
+
+                            nmos::receiver_state receiver_state(nmos::receiver_states::unknown);
+                            utility::string_t receiver_state_debug;
+
+                            const std::pair<nmos::id, nmos::type> connection_receiver_id_type{ receiver_id, nmos::types::receiver };
+                            auto connection_receiver = find_resource(connection_resources, connection_receiver_id_type);
+                            if (connection_resources.end() == connection_receiver) throw std::logic_error("Matching IS-05 receiver not found");
+
+                            auto& transport_file = nmos::fields::transport_file(nmos::fields::endpoint_staged(connection_receiver->data));
+
+                            if (validate_receiver)
```

Comment:
if `validate_receiver` is empty, should this use the "default" implementation, i.e. `nmos::experimental::validate_receiver_resources`? if it doesn't do any validation when this handler is omitted, the comment in `make_node_implementation_streamcompatibility_receiver_validator()` that "this example uses the default receiver validator explicitly" is wrong.

(same question applies to `validate_sender` above!)

Reply:
Added the default implementation for `validate_sender` but `gate` argument stopped me from making it for `validate_receiver`. Not sure yet if it's a real obstacle to do it.